ui/main/src/app/app.component.ts

```diff
@@ -168,14 +168,14 @@ export class AppComponent implements OnInit {
     if (!!config.i18n.supported.locales) {
       const localeRequests$ = [];
       (config.i18n.supported.locales as string[]).forEach(local => localeRequests$.push(this.i18nService.loadLocale(local)));
-      merge(...localeRequests$).pipe(skip(localeRequests$.length - 1)).subscribe(() => { // Wait for all request to complete
-        console.log(new Date().toISOString(), 'All opfab translation loaded for locales:', config.i18n.supported.locales.toString());
-        this.translateService.addLangs(config.i18n.supported.locales);
-        this.loadTranslationForMenu();
-        this.launchAuthenticationProcess();
+
+      localeRequests$.forEach(localeRequest$ => {
+          localeRequest$.subscribe(result => {
+            this.loadTranslationForMenu();
+          });
       });
+      this.launchAuthenticationProcess();
     }
```

Comment:
I have a doubt:  in this way we are not waiting for all the locales to be loaded before starting.  
Maybe keeping the merge() and removing ".pipe(skip(localeRequests$.length - 1))" would solve the issue.

Reply:
Yes, with my solution, we are not waiting for all locales to be loaded before starting.
The problem with the solution you propose is sometimes all languages are not completely loaded, with your solution as soon as we have one result the subscribe is finished.

Finally, I think we've found the solution, with the link you found : https://stackoverflow.com/questions/41734921/rxjs-wait-for-all-observables-in-an-array-to-complete-or-error.